cli/cli/commands/port/print/print.go

```diff
@@ -113,13 +131,32 @@ func run(
 		)
 	}
 
-	fullUrl := fmt.Sprintf("%v:%v", ipAddress, publicPort.GetNumber())
-	maybeApplicationProtocol := publicPort.GetMaybeApplicationProtocol()
-
-	if maybeApplicationProtocol != "" {
-		fullUrl = fmt.Sprintf("%v://%v", maybeApplicationProtocol, fullUrl)
+	fullUrl, err := formatOutput(format, ipAddress, publicPort)
+	if err != nil {
+		return stacktrace.Propagate(err, "Couldn't format the output according to formatting string '%v'", format)
 	}
-
 	out.PrintOutLn(fullUrl)
 	return nil
 }
+
+func formatOutput(format string, ipAddress string, spec *services.PortSpec) (string, error) {
+	parts := strings.Split(format, ",")
+	var resultParts []string
+	for _, part := range parts {
+		switch part {
+		case "protocol":
+			if spec.GetMaybeApplicationProtocol() != "" {
+				resultParts = append(resultParts, spec.GetMaybeApplicationProtocol()+"://")
+			} else {
+				logrus.Warnf("Expected protocol but was empty, skipping")
```

Comment:
Like the idea, but we should remove this else it's going to pollute the output and prevent users from doing things like `curl $(kurtosis port print) ...`

Reply:
Yeah this is pretty awkward. The idea that the default is to contain the application protocol that might not be present, leaves us on a spot where we cannot just ignore it, nor error it (since the default should never error IMO.

A workaround here would be to pipe stderr to `/dev/null` on `$()` which could be fine. Alternatively the user could specify the format that makes sense for the port (omitting the protocol if the protocol is indeed not present).

WDYT @mieubrisse ?

Reply:
Added this as a TODO item and moved to debug so we still have the log line if needed